src/cc_catalog_airflow/dags/util/etl/scripts/ExtractCCLinks.py

```diff
@@ -0,0 +1,284 @@
+
+from pyspark import SparkContext
+from pyspark.sql import SQLContext
+from pyspark.sql import SparkSession
+from pyspark.sql.types import *
+import boto3
+import botocore
+from botocore.handlers import disable_signing
+import gzip
+from urlparse import urlparse
+import requests
+from collections import Counter
+import random
+from warcio.archiveiterator import ArchiveIterator
+import ujson as json
+import StringIO
+import sys
+import re
+import logging
+from datetime import date, datetime, timedelta
+
+
+class CCLinks:
+
+    logging.getLogger('ExtractCCLinks')
+    logging.basicConfig(format='%(asctime)s: [%(levelname)s - ExtractCCLinks] =======> %(message)s', level=logging.INFO)
+
+
+    def __init__(self, _index, _ptn=2500):
+        """
+        CCLinks constructor: Validate the user-defined index based on Common Crawl's expected format.
+        If the pattern is valid, it generates 1) a url for the WAT path and 2) the location to output the results.
+
+        Parameters
+        ------------------
+        _index: string
+            The common crawl index name
+
+        _ptn: integer
+            The number of partitions for the spark job
+
+        Returns
+        ------------------
+        None
+
+        """
+
+        self.crawlIndex = _index
+
+        #check index format
+        pattern = re.compile('CC-MAIN-\d{4}-\d{2}')
+        if not pattern.match(_index):
+            logging.error('Invalid common crawl index format => {}.'.format(_index))
+            sys.exit()
+
+
+        self.numPartitions  = _ptn
+        self.url            = 'https://commoncrawl.s3.amazonaws.com/crawl-data/{}/wat.paths.gz'.format(self.crawlIndex)
+        self.output         = 's3://commonsmapper-v2/output/{}'.format(self.crawlIndex)
+
+
+    def loadWATFile(self):
+        #load the WAT file paths
+        """
+        Make a request for a WAT file using the url, that was defined in the constructor.
+
+        Parameters
+        ------------------
+        None
+
+        Returns
+        ------------------
+        list
+            A list of WAT path locations.
+        """
+        logging.info('Loading file {}'.format(self.url))
+
+        try:
+            response = requests.get(self.url)
+
+            if response.status_code == requests.codes.ok:
+                content     = StringIO.StringIO(response.content)
+                fh          = gzip.GzipFile(fileobj=content)
+                watPaths    = fh.read().split()
+
+                return watPaths
+            else:
+                raise Exception
+
+        except Exception as e:
+            logging.error('There was a problem loading the file.')
+            logging.error('{}: {}'.format(type(e).__name__, e))
+            #sys.exit()
+
+
+    def processFile(self, _iterator):
+        """
+        Parse each WAT file to identify domains with a hyperlink to creativecommons.org.
+
+        Parameters
+        ------------------
+       _iterator: iterator object
```

Comment:
It looks like all linting issues are on `src/cc_catalog_airflow/dags/util/etl/scripts/ExtractCCLinks.py`.  That file wasn't edited by this PR, only moved; it's extreemely non-PEP8-compliant.  It will be fixed in a future PR, but modifying that file is beyond the scope of this one, since a prerequisite is writing tests for that file.